sdk/core/Azure.Core.Experimental/src/Primitives/BinaryData.cs

```diff
@@ -39,8 +39,10 @@ public BinaryData(ReadOnlyMemory<byte> data)
         /// </summary>
         /// <param name="data">The string data.</param>
         /// <returns>A <see cref="BinaryData"/> instance.</returns>
```

Comment:
I really miss the implicit conversion to/from `ReadOnlyMemory<byte`.   With these renames, I think that we're making it clear that there is work required to convert to a string or to a stream.... but since the type name implies that this is binary, having the implicit conversion makes sense to me, semantically.

Reply:
This should still be there..